src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -171,114 +204,137 @@ private WordEmbeddingsTransform(IHost host, ModelLoadContext ctx, IDataView inpu
+
+            InputType = new VectorType(PrimitiveType.FromKind(DataKind.Text));
+            OutputType = new VectorType(NumberType.R4, 3 * _currentVocab.Dimension);
         }
 
-        public static WordEmbeddingsTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        public static WordEmbeddingsTransform Create(IHostEnvironment env, ModelLoadContext ctx)
         {
             Contracts.CheckValue(env, nameof(env));
             IHost h = env.Register(RegistrationName);
             h.CheckValue(ctx, nameof(ctx));
-            h.CheckValue(input, nameof(input));
             return h.Apply("Loading Model",
-                ch => new WordEmbeddingsTransform(h, ctx, input));
+                ch => new WordEmbeddingsTransform(h, ctx));
         }
 
+        // Factory method for SignatureLoadDataTransform.
+        public static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+            => Create(env, ctx).MakeDataTransform(input);
+
+        // Factory method for SignatureLoadRowMapper.
+        public static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
         public override void Save(ModelSaveContext ctx)
         {
             Host.CheckValue(ctx, nameof(ctx));
             ctx.CheckAtModel();
             ctx.SetVersionInfo(GetVersionInfo());
 
-            SaveBase(ctx);
+            base.SaveColumns(ctx);
             ctx.Writer.WriteBoolByte(_customLookup);
             if (_customLookup)
                 ctx.SaveString(_modelFileNameWithPath);
             else
                 ctx.Writer.Write((uint)_modelKind);
         }
 
-        protected override ColumnType GetColumnTypeCore(int iinfo)
+        protected override IRowMapper MakeRowMapper(ISchema schema)
+           => new Mapper(this, schema);
+
+        protected override void CheckInputColumn(ISchema inputSchema, int col, int srcCol)
         {
-            Host.Assert(0 <= iinfo && iinfo < Infos.Length);
-            return _outputType;
+            var colType = inputSchema.GetColumnType(srcCol);
+            if (!(colType is VectorType && colType.ItemType.IsText))
+                throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", ColumnPairs[col].input, "Text", inputSchema.GetColumnType(srcCol).ToString());
         }
 
-        protected override Delegate GetGetterCore(IChannel ch, IRow input, int iinfo, out Action disposer)
+        private sealed class Mapper : MapperBase
         {
-            Host.AssertValue(ch);
-            ch.AssertValue(input);
-            ch.Assert(0 <= iinfo && iinfo < Infos.Length);
-            disposer = null;
+            private readonly WordEmbeddingsTransform _parent;
 
-            var info = Infos[iinfo];
-            if (!info.TypeSrc.IsVector)
+            public Mapper(WordEmbeddingsTransform parent, ISchema inputSchema)
```

Comment:
>inputSchema [](start = 66, length = 11)

I think you need to validate somewhere this schema.
You can obtain transformer, but no one guarantee you what it's gonna be applied to same data, or even data with same schema.
We do have some in our base class, but we only check columns with specific names are present, but no one checks types.

Reply:
I think `GetOutputSchema` method in Estimator class does that. Is it not sufficient? or need to do it here has well?

---
In reply to: [217930424](https://github.com/dotnet/machinelearning/pull/928#discussion_r217930424) [](ancestors = 217930424)

Reply:
Fixed...

---
In reply to: [218168048](https://github.com/dotnet/machinelearning/pull/928#discussion_r218168048) [](ancestors = 218168048,217930424)